docs/limits.md

```diff
@@ -0,0 +1,258 @@
+
+This is a further small but meaningful step toward an actor-based model for agent internals.
+
+#### Ledger transaction management
+
+Currently, the framework does not manage any aspect of submitting multiple transactions to the ledgers. This responsibility is left to skills. Additionally, the ledger APIs/contract APIs take the ledger as a reference to determine the nonce for a transaction. If a new transaction is sent before a previous transaction has been processed then the nonce will not be incremented correctly for the second transaction. This can lead to submissions of multiple transactions with the same nonce, and therefore failure of subsequent transactions.
+
+A naive approach would involve manually incrementing the nonce and then submitting transactions into the pool with the correct nonce for eventual inclusion. The problem with this approach is that any failure of a transaction will cause non of the subsequent transactions to be processed for some ledgers (https://ethereum.stackexchange.com/questions/2808/what-happens-when-a-transaction-nonce-is-too-high). To recover from a transaction failure not only the failed transaction would need to be handled, but potentially also all subsequent transactions. It is easy to see that logic required to recover from a transaction failure early in a sequence can be arbitrarily complex (involving potentially new negotiations between agents, new signatures having to be generated etc.).
+
+A further problem with the naive approach is that it (imperfectly) replicates the ledger state (with respect to (subset of state of) a specific account).
+
+A simple solution looks as follows: each time a transaction is constructed (requiring a new nonce) the transaction construction is queued until all previous transactions have been included in the ledger or failed. This way, at any one time the agent has only at most one transaction pending with the ledger. Benefits: simple to understand and maintain, transaction only enter the mempool when they are ready for inclusion which has privacy benefits over submitting a whole sequence of transaction at once. Downside: at most one transaction per block.
+
+This approach is currently used and implemented across all the reference skills.
+
+Related, the topic of latency in transactions. State channels provide a solution. E.g. https://github.com/perun-network/go-perun. There could also be an interesting overlap with our protocols here.
+
+
+#### Unsolved problems in `Multiplexer` - `AgentLoop` interplay
+
+Problem 1: connection generates too many messages in a short amount of time, that are not consumed by the multiplexer
+Solution: Can be solved by slowing down connections receive method called, controlled by the inbox messages amount
+Side effects: Most of the connections should have an internal queue cause there is no synchronization between internal logic and multiplexer connection `receive` calls.
```

Comment:
```suggestion
Side effects: Most of the connections should have an internal queue because there is no synchronization between internal logic and multiplexer connection `receive` calls.
```